static/js/message_list_view.js

```diff
@@ -1445,4 +1445,103 @@ export class MessageListView {
             message_container.status_message = false;
         }
     }
+
+    /* This function exist for two purposes:
+        * To track the current `sticky_header` which have some different properties
+          like date being always displayed.
+        * Set date on message header corresponding to the message next to the header. */
+    update_sticky_recipient_headers() {
+        const rows_length = this._rows.size;
+        if (!rows_length) {
+            /* No headers are present */
+            return;
+        }
+        /* Intentionally remove sticky headers class here to make calculations simpler. */
+        $(".sticky_header").removeClass("sticky_header");
+        /* visible_top is navbar top position + height for us. */
+        const visible_top = message_viewport.message_viewport_info().visible_top;
+        /* We need date to be properly visible on the header, so partially visible headers
+           who are about to be scrolled out of view are not acceptable. */
+        const header_about_to_be_partially_hidden = visible_top - 1;
+
+        function is_sticky(header) {
+            const header_props = header.getBoundingClientRect();
+            const margin_between_recipient_rows = 10;
```

Comment:
This number should have comments next to its definition both here and in the CSS to ensure we keep the values in sync.

Reply:
Yeah, added comments.